src/services/authService.js

```diff
@@ -143,17 +144,59 @@ async function issueJWTToken(req, res) {
         return res.status(401).send("Unauthorized: expired token");
     }
 
-    //getting userinfo from db
-    const userInfo = await userModel.getUserById(tokenRecord.user_id);
+    //checking idp access token expiration time
+    if (Date.now() > Date.parse(tokenRecord.idp_access_token_expires_at)) {
+        //idp access token is expired
+        if (tokenRecord.idp_refresh_token) {
+            //idp refresh token is available
+            const oauthClient = await getOAuthClient();
+            try {
+                const newTokenSet = await oauthClient.refresh(
+                    tokenRecord.idp_refresh_token
+                );
+
+                if (!newTokenSet) {
+                    return res.status(401).send("Failed to refresh IDP token");
+                }
+
+                const newTokenRecord =
+                    await userTokenModel.updateIDPAccessToken(
+                        tokenRecord.user_id,
+                        newTokenSet
+                    );
+
+                // console.log(
```

Comment:
Let's clean up the `console.log` statements. For later, we can add a proper logger such as pino so that we can trace back any errors.